uefi/firmwarefile.go

```diff
@@ -69,6 +95,79 @@ type FirmwareFile struct {
 	buf    []byte
 }
 
+// Validate Firmware File
+func (f *FirmwareFile) Validate() []error {
+	errs := make([]error, 0)
+	buflen := uint64(len(f.buf))
+	blankSize := [3]byte{0xFF, 0xFF, 0xFF}
+	if buflen < FileHeaderMinLength {
+		errs = append(errs, fmt.Errorf("length too small!, buffer is only %#x bytes long", buflen))
+		return errs
+	}
+
+	// Size Checks
+	fh := &f.Header
+	if fh.Size == blankSize && buflen < FileHeaderExtMinLength {
```

Comment:
nit: you could group all the checks when the fh is blank under a single `if fh.Size == blankSize`, then in the body check for `buflen < FIleHeaderExtMinLength` and `!fh.Attributes.isLarge()` separately. Then an `else` for the other case.
It's more for readability (and to be future-proof in case of changes) but feel free to ignore